lib/access_import/access_import.rb

```diff
@@ -35,7 +44,16 @@ def make_volunteers
       ).any?
       volunteer_attrs = volunteer_transformer.prepare_attributes(ac_volunteer)
       volunteer = Volunteer.new(volunteer_attrs)
-      puts "Importing personen_rolle #{key} to Volunteer.id: #{volunteer.id}" if volunteer.save!
+      volunteer.created_at = ac_volunteer[:d_Rollenbeginn]
+      volunteer.updated_at = ac_volunteer[:d_MutDatum]
+      next unless volunteer.save!
+      message = "Importing personen_rolle #{key} to Volunteer.id: #{volunteer.id}"
+      if ac_volunteer[:d_Rollenende]
+        volunteer.state = Volunteer::RESIGNED
+        volunteer.save!
+        message += " -- Volunteer was set to resigned at #{ac_volunteer[:d_Rollenende]}  ------"
+      end
+      puts message
     end
     puts "Imported #{Volunteer.count - count_before_import} new volunteers from MS Access Datbase."
   end
```

Comment:
You should be able to DRY up this and extract a common method for ` make_volunteers` and `make_clients`.